controllers/replication-controller/dellcsireplicationgroup_controller.go

```diff
@@ -478,3 +516,340 @@ func (r *ReplicationGroupReconciler) SetupWithManager(mgr ctrl.Manager, limiter
+		if err != nil {
+			if errors.IsNotFound(err) {
+				log.V(common.InfoLevel).Info(fmt.Sprintf("PVC %s/%s deleted successfully", namespace, pvcName))
+				done = true
+			} else {
+				log.V(common.InfoLevel).Info(fmt.Sprintf("Error checking PVC %s/%s: %s", namespace, pvcName, err.Error()))
+			}
+		}
+		if iteration > 60 {
+			return fmt.Errorf("timed out waiting on PVC %s/%s to be deleted", namespace, pvcName)
+		}
+	}
+
+	// Swap some fields in the PVC.
+	localPV := pvc.Spec.VolumeName
+	pvc.Annotations[controller.RemotePV] = pvc.Spec.VolumeName
+	pvc.Spec.VolumeName = remotePV
+
+	remoteStorageClassName := pvc.Annotations[r.Domain+"/remoteStorageClassName"]
+	pvc.Annotations[r.Domain+"/remoteStorageClassName"] = *pvc.Spec.StorageClassName
+	pvc.Annotations[r.Domain+"/replicationGroupName"] = rgTarget
+	pvc.Labels[r.Domain+"/replicationGroupName"] = rgTarget
+	pvc.Spec.StorageClassName = &remoteStorageClassName
+	pvc.ObjectMeta.ResourceVersion = ""
+
+	// Re-create the PVC, now pointing to the target.
+	log.V(common.InfoLevel).Info(fmt.Sprintf("printing final PVC: %+v\n", pvc))
+	log.V(common.InfoLevel).Info(fmt.Sprintf("Recreating PVC %s", pvc.Name))
+	err = client.CreatePersistentVolumeClaim(ctx, pvc)
+	if err != nil {
+		log.V(common.InfoLevel).Info(fmt.Sprintf("Error creating final PVC: %s\n", err.Error()))
+		return err
+	}
+
+	// Update the target PV's claimref to point to our pvc.
+	pvcUID := pvc.ObjectMeta.UID
+	pvcResourceVersion := pvc.ObjectMeta.ResourceVersion
+	err = updatePVClaimRef(ctx, client, targetPV, pvc.Namespace, pvcResourceVersion, pvc.Name, pvcUID, log)
+	if err != nil {
+		return err
+	}
+
+	// Verify pvc is created and bound to new PVs
+	// remotePV is the current localPVName arg, localPV is the current remotePVName arg
+	err = verifyPVC(ctx, client, remotePV, localPV, pvcName, namespace, log)
+	if err != nil {
+		return err
+	}
+
+	// Remove the PVC reclaim of local PV
+	err = removePVClaimRef(ctx, client, localPV, namespace, pvcName, log)
+	if err != nil {
+		return err
+	}
+
+	// Restore the PVs original volume reclaim policy
+	err = setPVReclaimPolicy(ctx, client, pvc.Spec.VolumeName, remotePVPolicy, log)
+	if err != nil {
+		return err
+	}
+	err = setPVReclaimPolicy(ctx, client, pvc.Annotations[r.Domain+"/remotePV"], localPVPolicy, log)
+	if err != nil {
+		return err
+	}
+
+	log.V(common.InfoLevel).Info("Operation completed successfully")
+	return nil
+}
+
+func verifyPVC(ctx context.Context, client connection.RemoteClusterClient, localPVName string, remotePVName string, pvcName string, namespace string, log logr.Logger) error {
+	log.V(common.InfoLevel).Info("Verifying")
+	done := false
+	for iteration := 0; !done; iteration++ {
+		time.Sleep(1 * time.Second)
+		pvc, err := client.GetPersistentVolumeClaim(ctx, namespace, pvcName)
+
+		if (err == nil) && (localPVName == pvc.Spec.VolumeName) && (remotePVName == pvc.Annotations[controller.RemotePV]) {
+			done = true
```

Comment:
**suggestion:** It doesn't appear that `done` is needed, it isn't used anywhere.

For many of these waits, can we just return immediately instead of keeping track of done?